### PR TITLE
feature: Allow multi snapshotters

### PR DESCRIPTION
**Pull request by wangforthinker**
### Ⅰ. Describe what this PR did
Add a daemon config allow-multi-snapshotter, if set true, pouchd will allow multi snapshotter. 

### Ⅱ. Does this pull request fix one issue?

Fix https://github.com/alibaba/pouch/pull/2585

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)
Add test.

### Ⅳ. Describe how to verify it
Add a flag allow-multi-snapshotter, if set true, pouchd will ignore check multi snapshotter snapshots.

### CODE REVIEW

File: daemon/daemon_utils.go [hunk 1]
```diff
@@ -60,7 +60,7 @@ func initialRuntime(baseDir string, runtimes map[string]types.Runtime) error {
 }
 
 // checkSnapshotterValid checks whether the given snapshotter is valid
-func checkSnapshotterValid(snapshotter string, ctrdClient ctrd.APIClient) error {
+func checkSnapshotterValid(snapshotter string, ctrdClient ctrd.APIClient, allowMultiSnapshotter bool) error {
```
Comment (allencloud) [inline daemon/daemon_utils.go]: I think this function should be encapsulated into the package `ctrd` to be 
```
func (c *Client)  checkSnapshotterValid(snapshotter string,allowMultiSnapshotter bool) error
```

WDYT? @wangforthinker

**Comment by allencloud** [comment]
I left several comments. Wish to get your feedback. @wangforthinker

File: daemon/daemon_utils.go [hunk 2]
```diff
@@ -81,14 +81,17 @@ func checkSnapshotterValid(snapshotter string, ctrdClient ctrd.APIClient) error
...
-			return fmt.Errorf("failed to check snapshotter driver %s: %v", p.ID, err)
+		// if allowMultiSnapshotter, ignore check snapshots exist
+		if !allowMultiSnapshotter {
+			// check if other snapshotter exists snapshots
+			exist, err := checkSnapshotsExist(p.ID, ctrdClient)
+			if exist {
+				return fmt.Errorf("current snapshotter driver is %s, cannot change to %s", p.ID, snapshotter)
+			}
+
+			if err != nil {
```
[the remark this answers is not in the document]
Comment (wangforthinker) [reply daemon/daemon_utils.go]: Fix it !

File: daemon/daemon_utils.go [hunk 3]
```diff
@@ -81,14 +81,17 @@ func checkSnapshotterValid(snapshotter string, ctrdClient ctrd.APIClient) error
...
-		if exist {
-			return fmt.Errorf("current snapshotter driver is %s, cannot change to %s", p.ID, snapshotter)
-		}
-
-		if err != nil {
-			return fmt.Errorf("failed to check snapshotter driver %s: %v", p.ID, err)
+		// if allowMultiSnapshotter, ignore check snapshots exist
+		if !allowMultiSnapshotter {
+			// check if other snapshotter exists snapshots
+			exist, err := checkSnapshotsExist(p.ID, ctrdClient)
```
[the remark this answers is not in the document]
Comment (wangforthinker) [reply daemon/daemon_utils.go]: Fix it!

File: daemon/daemon_utils.go (hunk 1 above)
Comment (wangforthinker) [reply daemon/daemon_utils.go]: Fix it!

**Comment by allencloud** [comment]
Thanks for your update again. I removed the `DO-NOT-MERGE` label. In addition, after the update you made, I think it is necessary for @Ace-Tang to take a review again since the code change. WDYT? @Ace-Tang 
If it still satisfies you, I think we could make to make this move on to merge.

**Comment by Ace-Tang** [comment]
Look good to me, /cc @allencloud

**Comment by allencloud** [comment]
Thanks for your review. Merging... @Ace-Tang

**Comment by Ace-Tang** [comment]
PR should not be added to fixed, https://github.com/alibaba/pouch/pull/2585 was closed by error

**Pull request merged**